Stack/Core/Security/Certificates/CertificateValidator.cs

```diff
@@ -726,14 +891,40 @@ public bool GetIssuersWithChainSupportEnabled(X509Certificate2Collection certifi
             }
 
             // not a trusted issuer.
-            return null;
+            return (null, null);
+        }
+
+        /// <summary>
+        /// Returns the certificate information for a trusted issuer certificate.
+        /// </summary>
+        private CertificateIdentifier GetIssuer(
+            X509Certificate2 certificate,
+            CertificateIdentifierCollection explicitList,
+            CertificateStoreIdentifier certificateStore,
+            bool checkRecovationStatus)
+        {
+            // check for root.
+            if (X509Utils.IsSelfSigned(certificate))
+            {
+                return null;
+            }
+
+            (CertificateIdentifier result, ServiceResultException srex) =
+                GetIssuerNoException(certificate, explicitList, certificateStore, checkRecovationStatus
+                );
+            if (srex != null)
+            {
+                throw srex;
+            }
+            return result;
         }
 
         /// <summary>
         /// Throws an exception if validation fails.
         /// </summary>
         /// <param name="certificates">The certificates to be checked.</param>
         /// <exception cref="ServiceResultException">If certificate[0] cannot be accepted</exception>
+        [Obsolete("Class replaced by Opc.Ua.CertificateValidator.InternalValidate(X509Certificate2Collection certificates, ConfiguredEndpoint endpoint)")]
         protected virtual void InternalValidate(X509Certificate2Collection certificates)
         {
```

Comment:
old code without fixes is beng removed? Can you call into the new function here for back compat?